v3/src/data-interactive/handlers/log-message-handler.ts

```diff
@@ -1,8 +1,29 @@
+import { LoggableObject, LoggableValue, logMessageWithReplacement } from "../../lib/log-message"
+import { t } from "../../utilities/translation/translate"
 import { registerDIHandler } from "../data-interactive-handler"
-import { DIHandler, diNotImplementedYet } from "../data-interactive-types"
+import { DIHandler, DILogMessage, DIResources, DIValues,  } from "../data-interactive-types"
 
 export const diLogMessageHandler: DIHandler = {
-  notify: diNotImplementedYet
+  notify(resources: DIResources, values?: DIValues) {
+    const webviewModel = resources.interactiveFrame?.content
+    if (values) {
+      const {formatStr, replaceArgs} = values as DILogMessage
+      const message = formatStr ?? ""
+      const argsArray = replaceArgs != null
+        ? Array.isArray(replaceArgs) ? replaceArgs : [replaceArgs]
+        : undefined
+      const args: LoggableObject = {}
+      argsArray?.forEach((value: LoggableValue, index: number) => {
+        args[index.toString()] = value
+      })
+      webviewModel?.applyModelChange(()=>{},
+        { log: logMessageWithReplacement(message, args) }
+      )
+      return {success: true}
+    } else {
+      return {success: false, error: t("V3.DI.Error.valuesRequired")}
```

Comment:
```suggestion
      return valuesRequiredResult
```
I should have thought of this before. You can import `valuesRequiredResult` from `di-results.ts`.